### DISCUSSION TITLE
Index output collection for indexes other than map-reduce

### DISCUSSION THREAD

**Discussion by adambajguz** [ravendb/ravendb#19147] [Ideas]
Hello,
currently only map-reduce indexes can have an output to a collection. It would be nice to be able to output indexed data as a virtual collection for all index types.

**Why?**

Basically for the same reasons as for map-reduce indexes.
 - You can create recursive operations.
 - You can push to another location.
 - Subscription / Changes – so you can get notified when a value has been changed.
 
Raven's indexes are not like the indexes in other databases, they are powerful tools for data modeling, as presented for example in this article: https://ravendb.net/articles/using-machine-learning-with-ravendb

**Comment by ayende** [comment]
The issue with this is that map/reduce output to a collection _for a reason_, because they represent an aggregation of the data. 

If you want to take a collection and transform that (adding / filtering / etc) - you can use that using RavenDB ETL to the same server.
That would allow you to run the code to create new documents from existing ones.

**Comment by adambajguz** [comment]
@ayende RavenDB ETL is an out of process solution, which I assume will be much slower (network overhead). Thus, I still think that outputing to colection for non map-reduce indexes will be best, and actually may be a very powerful new feature of RavenDB.

**Reply by ayende** [reply]
It isn't actually out of process. In this case, it will generally target the same node and while it is using TCP, it actually will use in-memory transport.
In general, it is also *really* good in handling this without you realizing that there is anything in the background.

**Reply by adambajguz** [reply]
Oh, great. I didn't know about that.
